### PR TITLE
Add monitoring annotations and common labels

### PR DESCRIPTION
**Pull request by omegas27**

### CODE REVIEW

**Comment by omegas27** [comment]
Tested on ginger

File: helm/cluster-autoscaler-app/templates/service.yaml [hunk 1]
```diff
@@ -4,13 +4,20 @@ metadata:
   name: {{ .Values.name }}
   namespace: {{ .Values.namespace }}
   labels:
-    giantswarm.io/service-type: "{{ .Values.serviceType }}"
-    app: {{ .Values.name }}
+    {{- include "labels.common" . | nindent 4 }}
+    # kubernetes.io/cluster-service are kept to be displayed in kubectl cluster-info
     kubernetes.io/cluster-service: "true"
     kubernetes.io/name: {{ .Values.name }}
+    giantswarm.io/monitoring: "true"
```
Comment (sslavic) [inline helm/cluster-autoscaler-app/templates/service.yaml:11]: Shouldn't this be annotation?

File: helm/cluster-autoscaler-app/templates/service.yaml [hunk 2]
```diff
@@ -4,13 +4,20 @@ metadata:
...
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "8085"
+    prometheus.io/path: "/metrics"
 spec:
   selector:
-    app: {{ .Values.name }}
+    {{- include "labels.selector" . | nindent 4 }}
   ports:
   - name: "metrics"
     port: 8085
```
Comment (sslavic) [inline helm/cluster-autoscaler-app/templates/service.yaml:23]: Would it make sense to make the port configurable, and then reference variable value?

File: helm/cluster-autoscaler-app/templates/service.yaml (hunk 1 above)
Comment (omegas27) [reply helm/cluster-autoscaler-app/templates/service.yaml:11]: We decided to go with a label as explained: https://github.com/giantswarm/giantswarm/blob/12511-tc-prom-spec/content/docs/architecture-specs-adrs/specs/configuration-of-targets-in-tc-prometheus.md#labels-and-annotations

As this is not merged yet, we did not communicate this

Comment (sslavic) [reply helm/cluster-autoscaler-app/templates/service.yaml:11]: Sounds like there's some misalignment https://intranet.giantswarm.io/docs/monitoring/monitoring_of_apps/

//cc @JosephSalisbury @piontec

Comment (omegas27) [reply helm/cluster-autoscaler-app/templates/service.yaml:11]: The discussion was brought up https://github.com/giantswarm/giantswarm/pull/12512/files#r465583355

File: helm/cluster-autoscaler-app/templates/service.yaml (hunk 2 above)
Comment (omegas27) [reply helm/cluster-autoscaler-app/templates/service.yaml:23]: This could make sense on a new version :)
More customization is always a plus

**Pull request merged**